packages/flutter/test/widgets/scroll_position_test.dart

```diff
@@ -160,6 +229,110 @@ void main() {
     expect(count, 1);
   });
 
+  testWidgets('shrink listview', (WidgetTester tester) async {
+    // widget tests run at (800x600)@3.0x
+    await tester.pumpWidget(MaterialApp(
+      home: ListView.builder(
+        physics: const _JumpingScrollPhysics(),
+        itemBuilder: (BuildContext context, int index) => index == 0
+              ? const ExpandingBox(collapsedSize: 400, expandedsize: 1200)
+              : Container(height: 300, color: Colors.red),
+        itemCount: 2,
+      ),
+    ));
+
+    final ScrollPosition position =
+        tester.state<ScrollableState>(find.byType(Scrollable)).position;
+    expect(position.activity.runtimeType, IdleScrollActivity);
+    expect(position.minScrollExtent, 0.0);
+    expect(position.maxScrollExtent, 100.0);
+    expect(position.pixels, 0.0);
+    await tester.tap(find.byType(FlatButton));
+    await tester.pumpAndSettle();
+
+    final TestGesture drag1 = await tester.startGesture(const Offset(10.0, 500.0));
+    await tester.pumpAndSettle();
+    await drag1.moveTo(const Offset(10.0, 0.0));
+    await tester.pumpAndSettle();
+    await drag1.up();
+    await tester.pumpAndSettle();
+    expect(position.pixels, closeTo(500.00, 0.01));
```

Comment:
fwiw, `moreOrLessEquals` is a more practical way of doing `closeTo` when you are just trying to side-step floating point issues. (It doesn't require the second argument and will in due course be updated to actually do it mathematically correctly for floating point.)